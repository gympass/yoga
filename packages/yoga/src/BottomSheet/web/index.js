import BottomSheet from './BottomSheet';
import Header from './Header';
import Content from './Content';
import Footer from './Footer';

const ExportBottomSheet = Object.assign(BottomSheet, {
  Header,
  Content,
  Footer,
});

export default ExportBottomSheet;
