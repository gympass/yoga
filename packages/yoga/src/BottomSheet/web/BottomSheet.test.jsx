import React from 'react';
import { screen, render, fireEvent, cleanup } from '@testing-library/react';

import { ThemeProvider, Button } from '../..';

import BottomSheet from '.';

describe('<BottomSheet />', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { baseElement } = render(
      <ThemeProvider>
        <BottomSheet isOpen>
          <BottomSheet.Header>Title</BottomSheet.Header>
          <BottomSheet.Content>Subtitle</BottomSheet.Content>
          <BottomSheet.Footer>
            <Button secondary>Ok, got it</Button>
          </BottomSheet.Footer>
        </BottomSheet>
      </ThemeProvider>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should render a minimal bottom sheet', () => {
    const onActionMock = jest.fn();

    render(
      <ThemeProvider>
        <BottomSheet isOpen>
          <BottomSheet.Header>Title</BottomSheet.Header>
          <BottomSheet.Content>Subtitle</BottomSheet.Content>
          <BottomSheet.Footer>
            <Button onClick={onActionMock} secondary>
              Ok, got it
            </Button>
          </BottomSheet.Footer>
        </BottomSheet>
      </ThemeProvider>,
    );

    screen.getByText('Title');
    screen.getByText('Subtitle');

    const button = screen.getByRole('button', { name: /Ok, got it/i });

    fireEvent.click(button);

    expect(onActionMock).toHaveBeenCalledTimes(1);
  });

  it('should hide the close button', () => {
    render(
      <ThemeProvider>
        <BottomSheet isOpen onClose={() => {}}>
          <BottomSheet.Header>Title</BottomSheet.Header>
        </BottomSheet>
      </ThemeProvider>,
    );

    expect(screen.queryByRole('button')).toBeNull();
  });

  it('should show the close button', () => {
    render(
      <ThemeProvider>
        <BottomSheet isOpen onClose={() => {}}>
          <BottomSheet.Header onClose={() => {}}>Title</BottomSheet.Header>
        </BottomSheet>
      </ThemeProvider>,
    );

    expect(screen.getByRole('button')).not.toBeNull();
  });
});
