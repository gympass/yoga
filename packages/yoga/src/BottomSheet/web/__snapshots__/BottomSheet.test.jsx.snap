// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomSheet /> should match snapshot 1`] = `
.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 24px;
}

.c10 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #231B22;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c10:not([disabled]):hover,
.c10:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(35,27,34,0.45);
}

.c10:active {
  background-color: rgba(35,27,34,0.75);
  color: #FFFFFF;
}

.c10:active svg {
  fill: #FFFFFF;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 24px;
  line-height: 32px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  margin-bottom: 24px;
  text-align: center;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  margin-bottom: 32px;
  text-align: center;
}

.c1 {
  padding: 16px  16px 16px  16px;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
  border-radius: 8px;
  background-color: #FFFFFF;
}

.c2 {
  padding: 40px 32px 32px;
  width: 580px;
  min-height: 160px;
  border-radius: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 3;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: rgba(35,27,34,0.48);
}

.c3 {
  padding: 24px 20px 24px 20px;
  width: 100%;
  border-radius: 16px 16px 0 0;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  -webkit-animation: bottomsheet-slide-up;
  animation: bottomsheet-slide-up;
  -webkit-animation-duration: 400ms;
  animation-duration: 400ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c5 {
  color: #231B22;
  line-height: 24px;
  font-size: 16px;
  margin-bottom: 16px;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  text-align: left;
}

.c7 {
  font-weight: 400;
  color: #6B6B78;
  line-height: 20px;
  margin-bottom: 20px;
  font-size: 14px;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  text-align: left;
}

.c9 {
  margin-left: 0;
  gap: 16px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<body>
  <div />
  <div
    id="dialog"
  >
    <div>
      <div
        class="c0"
      >
        <div
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
        <section
          aria-modal="true"
          class="c1 c2 c3"
          data-focus-lock-disabled="false"
          role="dialog"
        >
          <h2
            class="c4 c5"
            tabindex="-1"
          >
            Title
          </h2>
          <p
            class="c6 c7"
          >
            Subtitle
          </p>
          <footer
            class="c8 c9"
            d="flex"
          >
            <button
              class="c10"
            >
              Ok, got it
            </button>
          </footer>
        </section>
        <div
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
      </div>
    </div>
  </div>
</body>
`;
